app/scripts/filters/util.js

```diff
@@ -528,6 +528,31 @@ angular.module('openshiftConsole')
       }
       var uri = new URI(url);
       var protocol = uri.protocol();
-      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');      
+      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');
+    };
+  })
+  // currently used to set an alt text value for an env var that has a
+  // valueFrom attribute instead of a value.
+  .filter('altTextForValueFrom', function() {
+    return function(obj) {
+      if(obj.value) {
+        return;
+      }
+      _.each(obj.valueFrom, function(val, key) {
```

Comment:
Expanding on my comment about names, I'd suggest something like

```js
_.each(obj.valueFrom, function(from, refType) { ...
```

then below you have lines like

```js
if (refType === 'configMapKeyRef') { ... }
```

and

```js
obj.valueAlt = 'Set to the key ' + from.key + ' in config map ' + from.name;
```

etc